File: Visual Studio LIghtswitch 2012/Chapter7/HelpDeskVB/HelpDeskVB/Client/UserCode/IssueDetailManager.vb
'You can use and redistribute the code from this book (and sample application) for non-commercial and 
'most commercial purposes as long as you acknowledge the source and authorship. 
'You should note the source of the code in any documentation as well as in the program code itself (as a comment). 
'The acknowledgment should include author, title, publisher, and ISBN. 
'An example of such an acknowledgment would be "Derived from Listing 2-10, LightSwitch 2012 by Tim Leung, published by Apress, ISBN 9781430250715".


Namespace LightSwitchApplication

    Public Class IssueDetailManager

        Private Sub Issue_Loaded(succeeded As Boolean)
            ' Write your code here.
            Me.SetDisplayNameFromEntity(Me.Issue)
        End Sub

        Private Sub Issue_Changed()
            ' Write your code here.
            Me.SetDisplayNameFromEntity(Me.Issue)
        End Sub

        Private Sub IssueDetailManager_Saved()
            ' Write your code here.
            Me.SetDisplayNameFromEntity(Me.Issue)
        End Sub

        'Listing 4-8. Resolving Conflicts in Code
        Private Sub IssueDetailManager_Saving(ByRef handled As Boolean)

            handled = True

            Try
                Me.DataWorkspace.ApplicationData.SaveChanges()
            Catch ex As ConcurrencyException

                For Each entityConflict In ex.EntitiesWithConflicts.OfType(Of Issue)()
                    entityConflict.Details.EntityConflict.ResolveConflicts(
                       Microsoft.LightSwitch.Details.ConflictResolution.ClientWins)
                Next

                Try
                    Me.DataWorkspace.ApplicationData.SaveChanges()
                    ShowMessageBox(
                       "Your record was modified by another user. Your changes have been kept.")

                Catch ex2 As Exception
                    ' A general exception has occurred 
                    ShowMessageBox(ex2.Message.ToString())
                End Try

            End Try
        End Sub

    End Class

End Namespace

File: Visual Studio LIghtswitch 2012/Chapter6/HelpDeskCS/HelpDeskCS/Client/UserCode/IssueDetailManager.cs
// You can use and redistribute the code from this book (and sample application) for non-commercial and 
// most commercial purposes as long as you acknowledge the source and authorship. 
// You should note the source of the code in any documentation as well as in the program code itself (as a comment). 
// The acknowledgment should include author, title, publisher, and ISBN. 
// An example of such an acknowledgment would be "Derived from Listing 2-10, LightSwitch 2012 by Tim Leung, published by Apress, ISBN 9781430250715".

using System;
using System.Linq;
using System.IO;
using System.IO.IsolatedStorage;
using System.Collections.Generic;
using Microsoft.LightSwitch;
using Microsoft.LightSwitch.Framework.Client;
using Microsoft.LightSwitch.Presentation;
using Microsoft.LightSwitch.Presentation.Extensions;

namespace LightSwitchApplication
{
    public partial class IssueDetailManager
    {
        partial void Issue_Loaded(bool succeeded)
        {
            // Write your code here.
            this.SetDisplayNameFromEntity(this.Issue);
        }

        partial void Issue_Changed()
        {
            // Write your code here.
            this.SetDisplayNameFromEntity(this.Issue);
        }

        partial void IssueDetailManager_Saved()
        {
            // Write your code here.
            this.SetDisplayNameFromEntity(this.Issue);
        }



        //Listing 4-8. Resolving Conflicts in Code
        partial void IssueDetailManager_Saving(ref bool handled)
        {

    //            handled = true;                                                                 

    //try
    //{
    //    DataWorkspace.ApplicationData.SaveChanges();                                
    //}
    //catch (ConcurrencyException ex)
    //{
    //    foreach (var entityConflict in ex.EntitiesWithConflicts.OfType<Issue>())    
    //    {
    //        conflictingStatus.EntityConflict.ResolveConflicts(
    //           Microsoft.LightSwitch.Details.ConflictResolution.ClientWins);        
    //    }

    //    try
    //    {
    //        this.DataWorkspace.ApplicationData.SaveChanges();
    //        this.ShowMessageBox(
    //          "Your record was modified by another user. Your changes have 
    //            been kept.");
    //    }
    //    catch (Exception ex2)
    //    {
    //       // A general exception has occurred 
    //        this.ShowMessageBox(ex2.Message.ToString());
    //     }
    //}


        }
    }
}